# Задача 7
## Условие задачи
Получены следующие результаты параллельных определений содержания антител в сыворотке крови (%):
2,15; 1,13; 2,04; 1,45; 1,35; 1,09;0,99; 2,07
Предполагая логнормальный закон распределения, найти среднее значение, СКО и доверительный интервал для 95% вероятности. Оценить результат анализа и его εr используя неизвестный закон распределения. Значимо ли отличаются результаты, вычисленные для различных законов распределения?


## Решение
### Логнормальное распределение
Посчитаем среднее логарифмов
$$\overline{\lg{x}}=\frac{\sum\limits_{i=1}^n\lg{x_i}}{n}=0.384$$
А также геометрическое среднее
$$X=e^{\overline{\lg{x}}}=1.469$$

In [7]:
import numpy as np
x = np.array([2.15, 1.13, 2.04, 1.45, 1.35, 1.09, 0.99, 2.07])
y = np.log(x)
mean_y = np.mean(y)
geom_mean_x = np.e ** mean_y
print(f'Среднее значение: {mean_y},\nСреднее геометричесоке: {geom_mean_x}')

Среднее значение: 0.3844974249084494,
Среднее геометричесоке: 1.4688759154567719


Вычислим среднее квадратическое отклонение
$$S_{\lg{x}}=\sqrt{\frac{1}{n-1}\sum\limits_{i=1}^{n}(\lg{x_i}-\lg{\bar{x}})^2}$$
Для измеряемой величины тогда среднее квадратическое отклонение будет равно
$$S_{\lg^{-1}{x}}=e^{S_{\lg{x}}}$$

In [8]:
S_lg=np.std(y, ddof=1)
S_x=np.e**S_lg
print(f'СКО для логарифма: {S_lg}')
print(f'СКО для измеряемой величины: {S_x}')

СКО для логарифма: 0.3142897346036469
СКО для измеряемой величины: 1.369286408734685


Доверительный интервал для уровня доверия P=0.95
$$\Delta_{0.95}=t_{0.95,7}\cdot \frac{S_{\lg{x}}}{\sqrt{n}}$$
Если логарифм лежит в интервале
$$\overline{\lg{x}}-t_{0.95,7}\cdot \frac{S_{\lg{x}}}{\sqrt{n}}\le\lg{x}\le\overline{\lg{x}}-t_{0.95,7}\cdot \frac{S_{\lg{x}}}{\sqrt{n}}$$
То измеряемое знаение
$$\frac{X}{e^{\Delta_{0.95}}}\le\lg{x}\le X\cdot e^{\Delta_{0.95}}$$

In [9]:

from scipy import stats

# Параметры
alpha = 0.05       # уровень значимости при 95%
df = len(x)-1            # число степеней свободы

# Критическое значение для двустороннего теста:
t = stats.t.ppf(1 - alpha/2, df)

print(f"t = {t:.4f}")

epsilon = t*S_lg/np.sqrt(len(x))
print(epsilon)

epsilon_down, epsilon_up = (geom_mean_x/np.e**epsilon, geom_mean_x*np.e**epsilon)
print(f'Нижняя граница: {epsilon_down},\nВерхняя граница: {epsilon_up}')

t = 2.3646
0.26275279358209824
Нижняя граница: 1.1294656348080794,
Верхняя граница: 1.9102807456161266


### Неизвестное распределение
Так как не знаем вид распределения, сделаем грубую оценку по неравенству Чебышева.
Среднее значение
$$\bar{x}=\frac{\sum\limits_{i=1}^n x_i}{n}$$
Средне квадратичное отклонение
$$S=\sqrt{\frac{1}{n-1}\sum_{i=1}^{n}(x_i-\bar{x})^2}$$
Доверительный интервал согласно нормальному распределнию
$$\Delta=u\cdot\frac{S}{\sqrt{n}}$$
$$P(u)=2\Phi(u)$$
$$u=\Phi^{-1}(\frac{P(u)}{2})=\Phi^{-1}(0.475)=1.95$$
Заметим, что часто берут значение $u=2$
Тогда
$$\epsilon=u\cdot\frac{S}{\sqrt{n}}$$

In [10]:
mean_x=np.mean(x)
S = np.std(x, ddof=1)
u=1.95
epsilon=u*S/np.sqrt(len(x))
upper_e=mean_x+epsilon
lower_e=mean_x-epsilon
print(f'Среднее значение: {mean_x}\nСКО: {S}\nДоверительный интервал: +-{epsilon}\n'
      f'Нижняя граница: {lower_e}\nВерхняя граница: {upper_e}')

Среднее значение: 1.53375
СКО: 0.481068379457937
Доверительный интервал: +-0.3316625454959022
Нижняя граница: 1.2020874545040978
Верхняя граница: 1.865412545495902


### Сравнение значений
| Параметр | Неизвестное распределение | Логнормальное распределение|
| --- | --- | --- |
| Среднее значение |1.53375 | 1.4688 |
| СКО | 0.481068379457937 | 1.369 |
| Нижняя граница | 1.1316 | 1.2021 |
| Верхняя граница | 1.9359 | 1.8654 |